### PR TITLE
Source 3: Add a few CAT tests to verify backwards compatibility

### PR DESCRIPTION
**Pull request by girarda**
## What
Add CATs to verify the backwards compatibility of
- Parquet files with Decimal fields
- CSV files with strings_can_be_null = False
- CSV files without a header and with rows to skip

### CODE REVIEW

File: airbyte-integrations/connectors/source-s3/acceptance-test-config.yml [hunk 1]
```diff
@@ -1,164 +1,185 @@
...
-      timeout_seconds: 1800
-    - config_path: secrets/jsonl_config.json
-      expect_records:
-        path: integration_tests/expected_records/jsonl.jsonl
-      timeout_seconds: 1800
-    - config_path: secrets/jsonl_newlines_config.json
-      expect_records:
-        path: integration_tests/expected_records/jsonl_newlines.jsonl
-      timeout_seconds: 1800
+      - config_path: secrets/config.json
```
Comment (girarda) [inline airbyte-integrations/connectors/source-s3/acceptance-test-config.yml:4]: the file was reformatted by precommit

**Comment by octavia-squidington-iii** [comment]
##  source-s3 test report (commit [`dd081d2884`](https://github.com/airbytehq/airbyte/pull/29368/commits/dd081d2884bc1552973bd47f77b2d835493f3e88)) - ❌

⏲️  Total pipeline duration: 13mn31s 

| Step                                                             | Result   |
|:-----------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-s3/metadata.yaml | ✅        |
| Connector version semver check                                   | ✅        |
| QA checks                                                        | ✅        |
| Code format checks                                               | ✅        |
| Connector package install                                        | ✅        |
| Build source-s3 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                       | ✅        |
| Integration tests                                                | ✅        |
| Acceptance tests                                                 | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/alex_more_s3_legacy_tests/1691773269/dd081d2884bc1552973bd47f77b2d835493f3e88/source-s3/3.1.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:dd081d2884bc1552973bd47f77b2d835493f3e88)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-s3 test
```

File: airbyte-integrations/connectors/source-s3/integration_tests/sample_files/csvfile/user_schema.csv [hunk 2]
```diff
@@ -0,0 +1,9 @@
+id,name,valid,valid_string,array,dict
```
Comment (girarda) [inline airbyte-integrations/connectors/source-s3/integration_tests/sample_files/csvfile/user_schema.csv:1]: [the test already existed](https://github.com/airbytehq/airbyte/pull/29368/files#diff-cb83352ec27d3b1f8ecfb27d063d2e28eb41aa712183871025840ca571132a6cL19). I should've added this file in a previous PR

**Comment by octavia-squidington-iii** [comment]
##  source-s3 test report (commit [`26e2cb99b2`](https://github.com/airbytehq/airbyte/pull/29368/commits/26e2cb99b27e5638350282a1bcd9df407f354f90)) - ✅

⏲️  Total pipeline duration: 10mn50s 

| Step                                                             | Result   |
|:-----------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-s3/metadata.yaml | ✅        |
| Connector version semver check                                   | ✅        |
| QA checks                                                        | ✅        |
| Code format checks                                               | ✅        |
| Connector package install                                        | ✅        |
| Build source-s3 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                       | ✅        |
| Integration tests                                                | ✅        |
| Acceptance tests                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/alex_more_s3_legacy_tests/1691774447/26e2cb99b27e5638350282a1bcd9df407f354f90/source-s3/3.1.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:26e2cb99b27e5638350282a1bcd9df407f354f90)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-s3 test
```

**Comment by octavia-squidington-iii** [comment]
##  source-s3 test report (commit [`e29e86b66d`](https://github.com/airbytehq/airbyte/pull/29368/commits/e29e86b66dbf181eeff8774617446d38378f9d1c)) - ✅

⏲️  Total pipeline duration: 10mn58s 

| Step                                                             | Result   |
|:-----------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-s3/metadata.yaml | ✅        |
| Connector version semver check                                   | ✅        |
| QA checks                                                        | ✅        |
| Code format checks                                               | ✅        |
| Connector package install                                        | ✅        |
| Build source-s3 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                       | ✅        |
| Integration tests                                                | ✅        |
| Acceptance tests                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/alex_more_s3_legacy_tests/1691778339/e29e86b66dbf181eeff8774617446d38378f9d1c/source-s3/3.1.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e29e86b66dbf181eeff8774617446d38378f9d1c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-s3 test
```

**Comment by octavia-squidington-iii** [comment]
##  source-s3 test report (commit [`d98431c1e1`](https://github.com/airbytehq/airbyte/pull/29368/commits/d98431c1e18bfc3997b81df7e95145cd1d9b3c9c)) - ✅

⏲️  Total pipeline duration: 10mn50s 

| Step                                                             | Result   |
|:-----------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-s3/metadata.yaml | ✅        |
| Connector version semver check                                   | ✅        |
| QA checks                                                        | ✅        |
| Code format checks                                               | ✅        |
| Connector package install                                        | ✅        |
| Build source-s3 docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                       | ✅        |
| Integration tests                                                | ✅        |
| Acceptance tests                                                 | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/alex_more_s3_legacy_tests/1691779121/d98431c1e18bfc3997b81df7e95145cd1d9b3c9c/source-s3/3.1.3/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d98431c1e18bfc3997b81df7e95145cd1d9b3c9c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-s3 test
```

**Review by brianjlai — APPROVED** [review]
🧪 !

**Pull request merged**